### ISSUE TITLE
Players sometimes add quest rewards to their wishlist rather than the actual quest items they need out of raids.

### ISSUE THREAD

**Issue by Lemmings19** [thatsmybis/thatsmybis#48]
This is listed as a good first issue not because it's a quick one, but because the logic should be relatively straightforward. @Lemmings19 has ideas for how we should and should not solve this, so ask him about it first.

Say I want [Avenger's Legguards](https://classic.wowhead.com/item=21390/avengers-legguards), and I put them on my wishlist. They require [Skin of the Great Sandworm](https://classic.wowhead.com/item=20931/skin-of-the-great-sandworm) from AQ40 to obtain.

When loot council is looking at Skin of the Great Sandworm, it won't show my wishlist item. Can we design and add a system that gracefully solves this mistake, or helps users avoid it?

This is a good first issue because it provides some opportunity to investigate different solutions and how they might impact the system. The proposed solution in the comments is a fairly simple one that only exists on the front-end and doesn't require knowledge of how the rest of the app works.

**Comment by Lemmings19** [comment]
tl;dr
I could write a warning message in Javascript that appears on the relevant pages when you enter an item that doesn't belong. Users could still find the item and wouldn't think there's a bug or it's missing. Other users could still add the item(s) if they wanted to. And everyone would get an early bit of education on avoiding making this mistake.

**Chat log:**
> [2:09 PM] teejay: One minor annoyance is you can put down both an item and its token (ie Eye of C'thun -> Ring of the Fallen God). Would be nice if you put down the item and the backend converted that to at token for loot awarding.
> [9:05 AM] Calaris: yeah we've been trying to get our players to just enter the token since that will be most useful for loot distribution
> [1:19 PM] Lemmings19: @teejay Tokens VS entering the token reward item is something that came up during development too. I was able to remove the duplicate Head of Onyxia, Nefarian, etc (Horde and Alliance each have their own copy of the item). I also added labels to a few items such as Thunderfury bindings (left) and (right).
> [1:20 PM] Lemmings19: I agree that this is a mistake people are going to make.
> [1:22 PM] Lemmings19: The solution that comes to mind involves a fair amount of work (creating a system for associating items with parent items, and doing the back-end conversion you spoke of). I'm a little bit skeptical about forcefully rewriting items that players chose. Perhaps a warning message would be a far easier solution, and it also wouldn't force the user's hand. If they really wanted to put that item down, they still could.
> [1:23 PM] Lemmings19: There are a number of other higher priority items waiting to be picked up too.
> [1:28 PM] Lemmings19: I've added this as a feature request. I wouldn't expect it to get picked up for a long time, unless another developer would like to tackle it. if I end up picking it up, I am likely to just add a warning message. https://github.com/thatsmybis/thatsmybis/issues/48
> GitHub
> Players sometimes add quest rewards to their wishlist rather than t...
> Say I want Avenger&#39;s Legguards, and I put them on my wishlist. They require Skin of the Great Sandworm from AQ40 to obtain. When loot council is looking at Skin of the Great Sandworm, it wo...
> 
> [1:32 PM] Lone Starr: could just take the final items out of the loot pool? can't imagine anyone running from the end result?
> [1:34 PM] Lemmings19: For the Loot Received section of the character profile, or if in the future we add a section for characters to list their current gear, we'll want it in the database. Perhaps a boolean or something added to the item table to filter it out (this removing them from the loot pool, but only for wishlisting/assigning loot).
> [1:34 PM] Lemmings19: That could work, and it would be pretty easy to implement. Good thinking!
> [1:35 PM] Lemmings19: There could be some weird edge case where some guild out there actually wants access to these items for wishlists and loot assignment. That's the only obvious downside that comes to mind.
> [1:35 PM] Lone Starr: what we have done is just not award it to anyone who doesnt have the token list, crude but working for now
> [1:36 PM] Lemmings19: And also confused users trying to add an item to their wishlist, but it's not showing up in the autocomplete... now I remember why I didn't look too far into adding this feature a month ago lol
> [1:36 PM] Lemmings19: I was worried about users trying to type stuff and it just not showing up.
> [1:37 PM] Lemmings19: I opted to just allow people to make mistakes... ones they'd quickly correct once they realized they weren't on the list.
> [1:38 PM] Lemmings19: I could write a warning message in Javascript that appears on the relevant pages when you enter an item that doesn't belong. Users could still find the item and wouldn't think there's a bug or it's missing. Other users could still add the item(s) if they wanted to. And everyone would get an early bit of education on avoiding making this mistake.
> [1:38 PM] Lemmings19: Does that sound reasonable?
> [1:39 PM] Lone Starr: yeah, a flag on the quest reward items could be a good temp measure
> [1:41 PM] Lemmings19: I'm also thinking that this way, I could just let the front-end deal with it and we wouldn't have to restructure any of the data or back-end functionality. It's a far less impactful change to the codebase and doesn't add any complexity. Just a little script that sits to the side on its own and doesn't interfere with anything.
